### PR TITLE
Feature/notifications

### PR DESCRIPTION
**Pull request by Domysee**
This also adds specific types for the player api.

### CODE REVIEW

File: bitmovin/notifications/emails.ts [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+} from '../utils/types';
+
+import {EmailNotificationWithConditions, EmailNotificationWithConditionsDetails} from './types';
+
+const emails = (configuration: InternalConfiguration, http: HttpClient = httpClient): NotificationEmails => {
+  const emailsBaseUrl = urljoin(configuration.apiBaseUrl, 'notifications', 'emails');
+  const encodingBaseUrl = urljoin(emailsBaseUrl, 'encoding');
+  const encodingsBaseUrl = urljoin(encodingBaseUrl, 'encodings');
+
+  const listAll = (limit, offset, sort, filter) => {
```
Comment (lukaskroepfl) [inline bitmovin/notifications/emails.ts]: there should be a list function builder utils function for that?

File: bitmovin/index.ts [hunk 2]
```diff
@@ -5,6 +5,7 @@ import analytics, {Analytics} from './analytics';
 import encoding, {Encoding} from './encoding';
 import player, {Player} from './player';
 
+import notifications, {Notifications} from './notifications/notifications';
```
Comment (lukaskroepfl) [inline bitmovin/index.ts]: would rather go with `notifications/index.ts` as we have it in the other dirs

File: bitmovin/notifications/emails.ts [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+export interface NotificationEmails {
+  list: List<EmailNotificationWithConditionsDetails>;
+  encoding: {
+    list: List<EmailNotificationWithConditionsDetails>;
+    encodings: {
+      (encodingId: string): {
+        liveInputStreamChanged: NotificationEmailsType;
+      };
+      liveInputStreamChanged: NotificationEmailsType;
```
Comment (lukaskroepfl) [inline bitmovin/notifications/emails.ts]: please also add `error` for encoding "errored"

File: bitmovin/notifications/types.ts [hunk 4]
```diff
@@ -1,6 +1,7 @@
 // email notifications
 export enum EventTypes {
-  LIVE_INPUT_STREAM_CHANGED = 'LIVE_INPUT_STREAM_CHANGED'
+  LIVE_INPUT_STREAM_CHANGED = 'LIVE_INPUT_STREAM_CHANGED',
+  ERROR = 'ERROR'
```
Comment (lukaskroepfl) [inline bitmovin/notifications/types.ts]: should be `ENCODING_ERROR`

File: package.json [hunk 5]
```diff
@@ -24,7 +24,7 @@
     "build": "yarn clean && yarn build:bundle:dev && yarn build:bundle:prod && yarn build:modules",
     "build:bundle:dev": "set NODE_ENV development webpack --config=./webpack4.config.js",
     "build:bundle:prod": "set NODE_ENV production webpack --config=./webpack4.config.js",
-    "build:modules": "tsc --project ./tsconfig.json && replace \"__VERSION__\" \"'$(node -p \"require('./package.json').version\")'\" ./dist/index.js",
+    "build:modules": "tsc --project ./tsconfig.json",
```
Comment (lukaskroepfl) [inline package.json]: `&& replace ...` needs to be there

File: bitmovin/notifications/types.ts (hunk 4 above)
Comment (Domysee) [reply bitmovin/notifications/types.ts]: This is done, not sure why github doesn't show this as outdated

**Pull request merged**